app/models/pet.rb

```diff
@@ -30,6 +30,8 @@
 class Pet < ApplicationRecord
   acts_as_tenant(:organization)
 
+  after_create { Organizations::DefaultPetTaskService.new(self).create_tasks }
+
```

Comment:
What do you think about explicitly calling this Service in the `PetsController#create` action rather than using a callback?

Callbacks can quickly become unwieldy and have some drawbacks. E.g. in tests, this callback will execute every time a Pet is created, even if the test doesn't require Tasks, leading to unnecessary database queries. This can considerably slow down the test suite as the number of tests grows.

Reply:
@Ptrboro calling on the controller create action is how I started writing this. I then decided to move it to the model as I thought it is something we would want to run every time. and living on the model it would save duplication in future should we ever create a pet in a different way. However, I agree that callbacks can become a mess, and for now calling the service on the create action is fine and probably a better approach. I will make this change.